### PR TITLE
Add support for on-demand feature flags

### PR DESCRIPTION
**Pull request by henrymercer**
This PR adds support for adding a new feature flag to the CodeQL Action without needing to make a corresponding change to an internal codebase.

Specifically, new features can be specified in a `features` query parameter delivered to the feature flags API.

While we're here, we also log feature flags on separate lines to avoid feature enablement being truncated.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/feature-flags.ts [hunk 1]
```diff
@@ -58,12 +58,24 @@ export enum Feature {
...
      * This overrides enablement status from the feature flags API.
      */
     envVar: string;
+    /**
+     * Whether the feature flag is part of the legacy feature flags API.
+     *
+     * These feature flags are included by default in the API response and do not need to be
+     * explicitly requested.
+     */
+    legacyApi?: boolean;
```
Comment (angelapwen) [inline src/feature-flags.ts:78]: Maybe we could include in the comments that the default value will be `false` if unspecified.

File: src/feature-flags.ts [hunk 2]
```diff
@@ -477,11 +477,15 @@ class GitHubFeatureFlags {
...
       );
-      const remoteFlags = response.data;
+      const remoteFlags = response.data as GitHubFeatureFlagsApiResponse;
       this.logger.debug(
-        "Loaded the following default values for the feature flags from the Code Scanning API: " +
-          `${JSON.stringify(remoteFlags)}`,
+        "Loaded the following default values for the feature flags from the Code Scanning API:",
       );
+      for (const [feature, value] of Object.entries(remoteFlags).sort(
+        ([nameA], [nameB]) => nameA.localeCompare(nameB),
```
Comment (angelapwen) [inline src/feature-flags.ts:485]: Nice to have these sorted!

**Review by angelapwen — DISMISSED** [review]

File: src/feature-flags.ts (hunk 1 above)
Comment (henrymercer) [reply src/feature-flags.ts]: Done, and resolved some merge conflicts!

**Review by angelapwen — APPROVED** [review]
:shipit:

**Pull request merged**